pkg/client/logs.go

```diff
@@ -173,15 +175,30 @@ func getPodsToStreamLogs(client kubernetes.Interface, cfg *LogConfig, podCh chan
 // or no plugin has been specified, retrieve all pods within the namespace. It will return an error if unable to create the watcher
 // but will continue to add pods to the channel in a separate go routine.
 func watchPodsToStreamLogs(client kubernetes.Interface, cfg *LogConfig, podCh chan *v1.Pod) error {
-	listOptions := metav1.ListOptions{}
+	var timeoutSeconds int64 = 5
+	listOptions := metav1.ListOptions{TimeoutSeconds: &timeoutSeconds}
 	if cfg.Plugin != "" {
 		selector := metav1.AddLabelToSelector(&metav1.LabelSelector{}, "sonobuoy-plugin", cfg.Plugin)
 		listOptions = metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(selector)}
 	}
 
-	watcher, err := client.CoreV1().Pods(cfg.Namespace).Watch(context.TODO(), listOptions)
+	lw := &cache.ListWatch{
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return client.CoreV1().Pods(cfg.Namespace).Watch(context.TODO(), listOptions)
```

Comment:
probably not to be addressed in this PR, but I'm wondering if we need to handle graceful cancellation here if user hits Ctrl+C / sends SIGINT, then we should cancel the context?